### PR TITLE
Add module variable to allow write permission

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request adds a module variable (`var.read_only`) that allows for the addition of write permissions to the IAM role being created.

> [!NOTE]
> This change is only possible because of cisagov/s3-read-role-tf-module#47.

It also loosens a version constraint to allow the module to work with a wider range of future versions of the AWS Terraform provider.

## 💭 Motivation and context ##

This change associated with write permissions is required because we require a read-write role that allows access to a particular COOL environment's remote state.

The version constraint change agrees with the advice given [here](https://developer.hashicorp.com/terraform/language/expressions/version-constraints#terraform-core-and-provider-versions).

Note also that these changes have been made in a way that is backward compatible with the role's previous behavior, so these are non-breaking changes.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert to the default branch of [cisagov/s3-read-role-tf-module](https://github.com/cisagov/s3-read-role-tf-module) once cisagov/s3-read-role-tf-module#49 is merged.

### CODE REVIEW

**Review by dv4harr10 — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
This looks great to me.  👍 👍 

My only thought is that with this change, the repo and module should now be renamed to something like `terraform-state-access-role-tf-module`, but I'm ok with just creating an issue for that change since it's a bit of a hassle.

File: variables.tf [hunk 1]
```diff
@@ -57,13 +57,19 @@ variable "create_assume_role" {
...
+variable "read_only" {
+  default     = true
+  description = "A Boolean value indicating whether or not to make the role and policy read-only.  If false then the role and policy will allow write permissions."
+  type        = bool
+}
+
 variable "role_description" {
-  default     = "Allows read-only access to the Terraform state at '%s' for the '%s' workspace(s) in the %s S3 bucket."
-  description = "The description to associate with the IAM role (as well as the corresponding policy) that allows read-only access to the specified state in the specified S3 bucket where Terraform state is stored.  Note that the first \"%s\" in this value will get replaced with the terraform_state_path variable, the second \"%s\" will get replaced with the terraform_workspace variable, and the third \"%s\" will get replaced with the terraform_state_bucket_name variable."
+  default     = "Allows %s access to the Terraform state at '%s' for the '%s' workspace(s) in the %s S3 bucket."
```
Comment (mcdonnnj) [inline variables.tf:71]: I don't see an update to `local.role_description` in this PR that reflects this change.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (jsf9k) [reply variables.tf:71]: Excellent catch!  See commit 52075df.

**Comment by jsf9k** [comment]
> This looks great to me. 👍 👍
> 
> My only thought is that with this change, the repo and module should now be renamed to something like `terraform-state-access-role-tf-module`, but I'm ok with just creating an issue for that change since it's a bit of a hassle.

Please see issue #38 (and cisagov/s3-read-role-tf-module#48).

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Comment by dav3r** [comment]
> Please see issue #38 (and [cisagov/s3-read-role-tf-module#48](https://github.com/cisagov/s3-read-role-tf-module/issues/48)).

Perfect, thanks! 😎

**Review by dv4harr10 — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Pull request merged**